src/plugins/data_source_management/framework/utils/shared.ts

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export function get<T = unknown>(obj: Record<string, any>, path: string, defaultValue?: T): T {
+  return path.split('.').reduce((acc: any, part: string) => acc && acc[part], obj) || defaultValue;
+}
+
+export function addBackticksIfNeeded(input: string): string {
+  if (input === undefined) {
+    return '';
+  }
+  // Check if the string already has backticks
+  if (input.startsWith('`') && input.endsWith('`')) {
+    return input; // Return the string as it is
+  } else {
+    // Add backticks to the string
+    return '`' + input + '`';
+  }
+}
+
+export function combineSchemaAndDatarows(
+  schema: Array<{ name: string; type: string }>,
+  datarows: Array<Array<string | number | boolean>>
+): object[] {
+  const combinedData: object[] = [];
+
+  datarows.forEach((row) => {
+    const rowData: { [key: string]: string | number | boolean } = {};
+    schema.forEach((field, index) => {
+      rowData[field.name] = row[index];
+    });
+    combinedData.push(rowData);
+  });
+
+  return combinedData;
+}
+
+export const formatError = (name: string, message: string, details: string) => {
+  return {
+    name,
+    message,
+    body: {
+      attributes: {
+        error: {
+          caused_by: {
+            type: '',
+            reason: details,
+          },
+        },
+      },
+    },
+  };
+};
+
+// TODO: relocate to a more appropriate location
+// Client route
+export const PPL_BASE = '/api/directquery/ppl';
```

Comment:
nit. I saw there are many repeated /api/directquery, we may extract them as a base constant.

Reply:
fixed